crypto/rsa/rsa_gen.c

```diff
@@ -445,6 +445,12 @@ static int rsa_keygen(OPENSSL_CTX *libctx, RSA *rsa, int bits, int primes,
             BN_clear_free(rsa->dmp1);
             BN_clear_free(rsa->dmq1);
             BN_clear_free(rsa->iqmp);
+            rsa->d = NULL;
+            rsa->p = NULL;
+            rsa->q = NULL;
+            rsa->dmp1 = NULL;
+            rsa->dmq1 = NULL;
+            rsa->iqmp = NULL;
```

Comment:
Why are these fields called 'intermediate results'?

Reply:
I think the term is applicable to these values in this context.. For this operation sensitive values must be cleared on failure. (This term comes from the specs somewhere)
Normally we free the RSA object when doing this type of operation.